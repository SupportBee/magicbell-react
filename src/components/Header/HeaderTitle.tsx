/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import Text from '../Text';

/**
 * Title for the header component.
 *
 * @example
 * <HeaderTitle />
 */
export default function HeaderTitle() {
  return (
    <p
      css={css`
        margin-right: 16px !important;
      `}
    >
      <Text id="header.title" defaultMessage="NOTIFICATIONS" />
    </p>
  );
}
