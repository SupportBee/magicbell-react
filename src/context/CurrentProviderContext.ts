import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
