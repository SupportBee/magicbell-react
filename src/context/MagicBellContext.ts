import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
