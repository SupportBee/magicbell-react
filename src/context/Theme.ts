export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
