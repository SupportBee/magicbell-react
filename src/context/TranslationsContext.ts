import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
