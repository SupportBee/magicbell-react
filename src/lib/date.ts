import dayjs from 'dayjs';
import localizedFormat from 'dayjs/plugin/localizedFormat';
import relativeTime from 'dayjs/plugin/relativeTime';
import updateLocale from 'dayjs/plugin/updateLocale';

dayjs.extend(localizedFormat);
dayjs.extend(relativeTime);
dayjs.extend(updateLocale);

dayjs.updateLocale('en', {
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: '%ds',
    m: '1m',
    mm: '%dm',
    h: '1h',
    hh: '%dh',
    d: '1d',
    dd: '%dd',
    M: '1mo',
    MM: '%dmo',
    y: '1y',
    yy: '%dy',
  },
});
