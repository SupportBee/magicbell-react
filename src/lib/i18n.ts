import dayjs from 'dayjs';
import 'dayjs/locale/es';
import 'dayjs/locale/pt-br';
import defaultsDeep from 'lodash/defaultsDeep';
import es from './translations/es';
import pt_br from './translations/pt-br';

export type CustomLocale = {
  name: string;
  translations: any;
};

function setDayjsLocale(localeName: string) {
  // All dayjs files are in lowercase and use hypens instead of underscores.
  const standardLocaleName = localeName.toLowerCase().replace('_', '-');
  dayjs.locale(standardLocaleName);
}

/**
 * Function to build localization data for the `TranslationsProvider`.
 *
 * @param locale
 */
export function useLocale(locale: string | CustomLocale) {
  const localeName = typeof locale === 'string' ? locale : locale.name;
  setDayjsLocale(localeName);

  const defaultLocales = { es, pt_BR: pt_br };

  if (typeof locale === 'string') return defaultLocales[locale];
  else {
    let customLocales = {};
    defaultsDeep(customLocales, { [localeName]: locale.translations }, defaultLocales);
    return customLocales[localeName];
  }
}
