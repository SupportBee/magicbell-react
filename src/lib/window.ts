/* eslint-disable no-restricted-globals */
import { useEffect, useRef, useState } from 'react';
import ResizeObserver from 'resize-observer-polyfill';

/**
 * Open a centered popup window.
 *
 * @param url Url of the floating window
 */
export function openWindow(url: string) {
  const top = (screen.height - 400) / 4;
  const left = (screen.width - 600) / 2;
  window.open(url, '', `width=600,height=400,scrollbars=no,top=${top},left=${left}`);
}

/**
 * Custom hook to observe the height of an HTML element.
 *
 * @param ref
 * @param initialHeight
 */
export function useHeight(ref, initialHeight) {
  const [height, setHeight] = useState(initialHeight);
  const resizeObserverRef = useRef() as any;

  useEffect(() => {
    resizeObserverRef.current = new ResizeObserver((entries: any = []) => {
      entries.forEach((entry) => {
        setHeight(entry.contentRect.height);
      });
    });

    if (ref.current) resizeObserverRef.current.observe(ref.current);

    return () => {
      if (resizeObserverRef.current) resizeObserverRef.current.disconnect();
    };
  }, [ref]);

  return height;
}
