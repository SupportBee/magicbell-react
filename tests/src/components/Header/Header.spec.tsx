import { render, RenderResult, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Header from '../../../../src/components/Header';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';
import { TranslationsProvider } from '../../../../src/context/TranslationsContext';
import { useLocale } from '../../../../src/lib/i18n';

describe('components', () => {
  describe('Header', () => {
    let onAllRead: () => void;
    let view: RenderResult;

    beforeEach(() => {
      onAllRead = jest.fn();
      const definitions = useLocale('es');

      view = render(
        <TranslationsProvider value={definitions}>
          <MagicBellThemeProvider value={defaultTheme}>
            <Header onAllRead={onAllRead} />
          </MagicBellThemeProvider>
        </TranslationsProvider>,
      );
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      it('renders a title and a button to mark all notification as read', () => {
        expect(view.container).toMatchSnapshot();
      });

      describe('default language', () => {
        it('renders a title in english', () => {
          view.rerender(
            <MagicBellThemeProvider value={defaultTheme}>
              <Header onAllRead={onAllRead} />
            </MagicBellThemeProvider>,
          );

          expect(view.container).toMatchSnapshot();
        });
      });
    });

    describe('.handleClick', () => {
      it('invokes the onAllRead callback', async () => {
        userEvent.click(view.getByTestId('mark-all-as-read'));
        await waitFor(() => expect(onAllRead).toBeCalledTimes(1));
      });
    });
  });
});
