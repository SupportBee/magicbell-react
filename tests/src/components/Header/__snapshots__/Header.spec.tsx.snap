// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header render default language renders a title in english 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgb(52, 152, 244)!important;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px!important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  text-align: left!important;
  text-transform: none!important;
  padding: 16px 24px!important;
  -webkit-flex-shrink: 0!important;
  -ms-flex-negative: 0!important;
  flex-shrink: 0!important;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:first-of-type {
  border-radius: 8px 8px 0 0!important;
}

.emotion-0:last-of-type {
  border-radius: 0 0 8px 8px!important;
}

.emotion-1 {
  margin-right: 16px!important;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      NOTIFICATIONS
    </p>
    <button
      data-testid="mark-all-as-read"
    >
      Mark All Read
    </button>
  </div>
</div>
`;

exports[`components Header render renders a title and a button to mark all notification as read 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgb(52, 152, 244)!important;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px!important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  text-align: left!important;
  text-transform: none!important;
  padding: 16px 24px!important;
  -webkit-flex-shrink: 0!important;
  -ms-flex-negative: 0!important;
  flex-shrink: 0!important;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:first-of-type {
  border-radius: 8px 8px 0 0!important;
}

.emotion-0:last-of-type {
  border-radius: 0 0 8px 8px!important;
}

.emotion-1 {
  margin-right: 16px!important;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      NOTIFICACIONES
    </p>
    <button
      data-testid="mark-all-as-read"
    >
      Marcar todo como le√≠do
    </button>
  </div>
</div>
`;
